python/examples/querying-all-installed-packages.py

```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/python3
+
+# Query all installed packages.
+# A Python equivalent for `rpm -qa`
+
+import rpm
+
+ts = rpm.TransactionSet()
+mi = ts.dbMatch()
+for hdr in mi:
```

Comment:
This isn't wrong in such a small example of course, but assigning the match iterator to a variable can leave the database cursor open for much longer than intended. With read-cursors it's not a big deal with in the post BerkeleyDB world, but when you don't actually need the match iterator handle outside the loop (ie the usual case), I'd recommend this style instead:
```
for hdr in ts.dbMatch(...):
    # do stuff
```

Reply:
Similar issues exist with the 'ts' handle of course - doesn't matter at all in such small examples but people commonly don't realize that the 'ts' handle will also keep the database open for much longer than intended.